cmd/argocd/commands/admin/backup_test.go

```diff
@@ -362,6 +372,430 @@ status: {}
+			args: args{
+				bak: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-cm
+  namespace: argo-cd
+  labels:
+    env: dev
+data:
+  foo: bar
+`,
+				live: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-cm
+  namespace: argo-cd
+  labels:
+    env: dev
+data:
+  foo: bar
+`,
+			},
+			applicationNamespaces:    []string{"argo-*"},
+			applicationsetNamespaces: []string{"argo-*"},
+		},
+		{
+			name: "It should create live resource if it is missing",
+			args: args{
+				bak: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-cm
+  namespace: argocd
+  labels:
+    env: dev
+data:
+  foo: bar
+`,
+				live: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-cm
+  namespaces: argocd
+  labels:
+    env: dev
+`,
+			},
+			applicationNamespaces:    []string{"argocd", "dev"},
+			applicationsetNamespaces: []string{"argocd", "prod"},
+		},
+		{
+			name: "It should prune live resources not present in backup when prune is enabled",
+			args: args{
+				bak: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: configmap-to-keep
+  namespace: default
+`,
+				live: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: configmap-to-keep
+  namespace: default
+  labels:
+    env: dev
+data:
+  foo: bar
+`,
+			},
+			prune: true,
+		},
+		{
+			name: "It should clear the operation field when stopOperation is true",
+			args: args{
+				bak: `apiVersion: v1
+kind: Application
+metadata:
+  name: app
+  namespace: argocd
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/example/repo.git
+    path: .
+    targetRevision: HEAD
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: default
+`,
+				live: `apiVersion: v1
+kind: Application
+metadata:
+  name: app
+  namespace: argocd
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/example/repo.git
+    namespace: default
+status:
+  operationState:
+    phase: Running
+    operation:
+      sync:
+        revision: HEAD
+`,
+			},
+			stopOperation: true,
+		},
+		{
+			name: "It should override live object when --override-on-conflict is true",
+			args: args{
+				bak: `apiVersion: v1
+kind: Secret
+metadata:
+  name: my-secret
+  namespace: argocd
+  labels:
+    env: dev
+type: Opaque
+data:
+  username: bar
+  password: abc
+`,
+				live: `apiVersion: v1
+kind: Secret
+metadata:
+  name: my-secret
+  namespace: argocd
+  labels:
+    env: prod
+type: Opaque
+data:
+  username: old
+  password: old-pwd
+`,
+			},
+			applicationNamespaces:    []string{"argocd"},
+			applicationsetNamespaces: []string{},
+			overrideOnConflict:       true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			bakObj := decodeYAMLToUnstructured(t, tt.args.bak)
+			liveObj := decodeYAMLToUnstructured(t, tt.args.live)
+			pruneObjects := make(map[kube.ResourceKey]unstructured.Unstructured)
+
+			configMap := &unstructured.Unstructured{}
+			configMap.SetUnstructuredContent(map[string]any{
```

Comment:
```suggestion
			configMap.SetUnstructuredContent(map[string]any{
```
why is this required?

Reply:
@nitishfy , it's being used [here](https://github.com/argoproj/argo-cd/pull/22780/files#diff-b2b9976573c508d94790348115fc627f48fabac7e518863624e40ab55180a86eR693).